### ISSUE TITLE
dataclasses: can't call a standalone function class field through an instance

### ISSUE THREAD

**Issue by bluetech** [microsoft/pyright#4432]
**Describe the bug**

If a standalone function (not a method) is assigned to a class variable in a dataclass, and called through an instance of the class using dot-notation, pyright doesn't understand that the instance is passed implicitly as the first argument.

**To Reproduce**

```py
# pyright: strict
import dataclasses

def standalone(obj: object) -> None:
    print(obj)

@dataclasses.dataclass
class Dataclass:
    not_a_method = standalone

Dataclass().not_a_method()
```

**Expected behavior**

This code runs fine at runtime. I think pyright should not complain about it, but it emits this error:

```
x.py:11:1 - error: Argument missing for parameter "obj" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**

pyright command line version 1.1.287

**Additional context**

I tried to reduce this to something not involving dataclasses, but it type-checks fine if `@dataclasses.dataclass` is commented out.

I tried changing to `not_a_method: ClassVar = standalone` but same error.

My real use case: `replace = dataclasses.replace`, so I can use `obj.replace(field=10)` like in named tuples, instead of using the standalone function.

`mypy --strict` doesn't complain on this.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.289, which I just published. It will also be included in a future release of pylance.
